### DISCUSSION TITLE
[22 July 2024] Spark.NET for .NET 9.0?

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [dotnet/spark#1160] [General]
# Third party to Official library in .NET 9.0
Migration from a third party library to **official** [**System.Runtime.Serialization.Formatters**](https://www.nuget.org/packages/System.Runtime.Serialization.Formatters/9.0.0-preview.6.24327.7)

<details>
<summary><strong>[15th July 2024]</strong>Spark.NET 3.5.1 for .NET 8.0 with <strong>No MessagePack serialization</strong></summary>

![image](https://github.com/user-attachments/assets/a477555d-3863-4f3a-a73b-51bd635ba4fd)

```bash
$ spark-shell --version
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 3.5.1
      /_/

Using Scala version 2.12.18, Java HotSpot(TM) 64-Bit Server VM, 1.8.0_211
Branch HEAD
Compiled by user heartsavior on 2024-02-15T11:24:58Z
Revision fd86f85e181fc2dc0f50a096855acf83a6cc5d9c
Url https://github.com/apache/spark
Type --help for more information.
```

For detail => https://github.com/dotnet/spark/pull/1178

- Extracted binary formatter in a separate class + tests

https://github.com/grazy27/spark/blob/main/docs/how-does-it-work.md
![image](https://github.com/user-attachments/assets/6d9fb9c0-f6dd-4ebc-992c-b128e0d9fa56)
![image](https://github.com/user-attachments/assets/a48adf6d-4f39-4cf3-a77e-f29f8eaf13cc)
</details>

<details>
<summary><strong>[16th April 2024]</strong>Spart.Net 3.2 for .NET 8.0 with <strong>MessagePack serialization</strong></summary>

https://github.com/dbeavon/sparkdotnet

## PR:[ Replace BinaryFormatter with MessagePack serialization](https://github.com/dotnet/spark/pull/1166)
using a **third party** library: https://github.com/MessagePack-CSharp/MessagePack-CSharp

</details>

<details>
<summary><strong>[30thAug2023]</strong> .NET for Apache Spark documentation will no longer be updated</summary>

[30thAug2023] .NET for Apache Spark documentation will no longer be updated

We're no longer updating the .NET for Apache Spark documentation, and have [archived](https://github.com/dotnet/docs/issues/34013#issuecomment-1699623332) it to https://learn.microsoft.com/previous-versions/dotnet/spark/what-is-spark. As a result, we're closing all the active .NET for Apache Spark issues. I'm sorry we couldn't address your issue.

</details>

**Comment by grazy27** [comment]
I'll create a PR when my prev one is merged

**Reply by grazy27** [reply]
Since .NET9 is not LTS, it makes sense to wait for .net10 to upgrade

![image](https://github.com/user-attachments/assets/33bf1b64-53ef-4df1-91ca-cdf108398f06)

**Comment by wudanzy** [comment]
Agreed! Let's wait until .NET 10.
